10_temperature_object/temperature.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class Temperature
+  class << self
+    def from_fahrenheit(temp)
+      Temperature.new(fahrenheit: temp)
+    end
+
+    def from_celsius(temp)
+      Temperature.new(celsius: temp)
+    end
+  end
+
+  def initialize(options = {})
+    @fahrenheit = options[:fahrenheit]
+    @celsius = options[:celsius]
+  end
+
+  def in_fahrenheit
+    return @fahrenheit if @fahrenheit
+
+    (@celsius * (9.0 / 5.0)) + 32
```

Comment:
`attr_reader` is your friend forever, dude. Avoid accessing instance variables in classes